test/node-rpc-test.js

```diff
@@ -0,0 +1,230 @@
+  apiKey: 'foo',
+  timeout: 15000
+});
+
+const wclient = new WalletClient({
+  port: ports.wallet,
+  apiKey: 'foo'
+});
+
+describe('RPC', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+  });
+
+  after(async () => {
+    await nclient.close();
+    await node.close();
+  });
+
+  it('should rpc help', async () => {
+    assert(await nclient.execute('help', []));
+    await assert.rejects(async () => {
+      await nclient.execute('help', ['getinfo']);
+    }, {
+      name: 'Error',
+      message: /^getinfo/
+    });
+  });
+  it('should rpc getinfo', async () => {
+    const info = await nclient.execute('getinfo', []);
+    assert.strictEqual(info.blocks, 0);
+  });
+
+  describe('getblock', function () {
+    it('should rpc getblock', async () => {
+      const {chain} = await nclient.getInfo();
+      const info = await nclient.execute('getblock', [chain.tip]);
+
+      const properties = [
+        'hash', 'confirmations', 'strippedsize',
+        'size', 'weight', 'height', 'version',
+        'versionHex', 'merkleroot', 'coinbase',
+        'tx', 'time', 'mediantime', 'nonce',
+        'bits', 'difficulty', 'chainwork',
+        'nTx', 'previousblockhash', 'nextblockhash'
+      ];
+
+      for (const property of properties)
+        assert(property in info);
+
+      assert.deepEqual(chain.height, info.height);
+      assert.deepEqual(chain.tip, info.hash);
+    });
+
+    it('should return correct height', async () => {
+      // Create an address to mine with.
+      const wallet = await node.plugins.walletdb.wdb.get(0);
+      const key = await wallet.createReceive(0);
+      const address = key.getAddress().toString(node.network.type);
+
+      // Mine two blocks.
+      await nclient.execute('generatetoaddress', [2, address]);
+
+      const {chain} = await nclient.getInfo();
+      const info = await nclient.execute('getblock', [chain.tip]);
+
+      // Assert the heights match.
+      assert.deepEqual(chain.height, info.height);
+    });
+
+    it('should return confirmations (main chain)', async () => {
+      const {chain} = await nclient.getInfo();
+
+      const {genesis} = node.network;
+      const hash = genesis.hash.toString('hex');
+
+      const info = await nclient.execute('getblock', [hash]);
+
+      assert.deepEqual(chain.height, info.confirmations - 1);
+    });
+
+    it('should return confirmations (orphan)', async () => {
+      // Get the current chain state
+      const {chain} = await nclient.getInfo();
+
+      // Get the chain entry associated with
+      // the genesis block.
+      const {genesis} = node.network;
+      let entry = await node.chain.getEntry(genesis.hash);
+
+      // Reorg from the genesis block.
+      for (let i = 0; i < chain.height + 1; i++) {
+        const block = await node.miner.mineBlock(entry);
+        await node.chain.add(block);
+        entry = await node.chain.getEntry(block.hash());
+      }
```

Comment:
I know this is how the test was merged into bcoin, but you don't need a reorg for this: I think you can just mine a single alternate block off of any block in the current chain that isn't the tip. It will be saved as an alternate and should not have a main chain height.

Reply:
Good point. I would like to add in a call to `getblock` with the tip before it is reorged out to show that the same block shows confirmations and then is -1 after the reorg. What you are describing could be another test case.